### PR TITLE
[python] Add schema short-circuit to SplitRead and FileScanner read paths

### LINKED ISSUE #8216: [Bug][python] SplitRead redundantly reads schema from filesystem when current schema is already in memory
### Search before asking

- [x] I searched in the [issues](https://github.com/apache/paimon/issues) and found nothing similar.

### Paimon version

master (latest)

### Compute Engine

PythonAPI

### Minimal reproduce step

**Minimal reproduce step:**
```markdown
1. Create a Paimon table and write data
2. Use pypaimon to read data via `SplitRead`
3. Observe that `schema_manager.get_schema()` is called even when `schema_id` matches the current table schema id
4. This triggers redundant filesystem reads for schema files that are already available in memory

### What doesn't meet your expectations?

When `schema_id == table.table_schema.id`, the Python read path should return the in-memory `table.table_schema` directly without filesystem access, matching the Java short-circuit pattern in `RawFileSplitRead.createFileReader()`.

### Anything else?

This is a companion fix to `FileScanner._schema_fields` which had the same redundant read pattern. Both share the root cause: missing short-circuit for the current table schema id.

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by MgjLLL**
### Purpose

Fix redundant filesystem I/O in `SplitRead` and `FileScanner` when reading schema.

`SplitRead` has 3 call sites that unconditionally call `schema_manager.get_schema(schema_id)` even when `schema_id == table.table_schema.id` — the schema is already in memory. This causes unnecessary filesystem reads in the common case (no schema evolution).

Java equivalent (`RawFileSplitRead.createFileReader()`) short-circuits with:
```java
schemaId == schema.id() ? schema : schemaManager.schema(schemaId)
```

### Changes

- `split_read.py`: Add `_resolve_schema()` method that returns in-memory schema when id matches, replacing 3 direct `get_schema()` calls in `raw_reader_supplier`, `_get_fields_and_predicate`, and `_file_read_fields`
- `file_scanner.py`: Add `_schema_fields()` method with same short-circuit pattern for `SimpleStatsEvolutions`

### Tests

- Added `file_scanner_schema_fields_test.py` with 3 test cases covering short-circuit, delegation, and zero-id edge case
- All existing tests pass (106 passed)

This closes #8216

### CODE REVIEW

File: paimon-python/pypaimon/read/scanner/file_scanner.py [hunk 1]
```diff
@@ -227,14 +227,19 @@ def __init__(
         # bucket set per ``total_buckets`` value.
         self._bucket_selector = self._init_bucket_selector()
 
-        def schema_fields_func(schema_id: int):
-            return self.table.schema_manager.get_schema(schema_id).fields
-
         self.simple_stats_evolutions = SimpleStatsEvolutions(
-            schema_fields_func,
+            self._schema_fields,
```
Comment (JingsongLi) [inline paimon-python/pypaimon/read/scanner/file_scanner.py:231]: This only protects the SimpleStatsEvolutions callback, but a normal scan still resolves the file schema while decoding manifest entries before _filter_manifest_entry() can use this helper. ManifestFileManager.read() still calls table.schema_manager.get_schema(file_dict["_SCHEMA_ID"]).fields for every entry, so current-schema files from a REST catalog can still hit the same 403 before this short-circuit is reached. Please apply the same current-schema short-circuit in the manifest decode path as well, or pass a resolver into ManifestFileManager.

File: paimon-python/pypaimon/read/split_read.py [hunk 2]
```diff
@@ -167,6 +167,14 @@ def _compute_nested_path_by_name(self) -> Optional[Dict[str, List[str]]]:
     def _nested_path_by_name(self) -> Optional[Dict[str, List[str]]]:
         return self._cached_nested_path_by_name
 
+    def _resolve_schema(self, schema_id: int):
```
Comment (JingsongLi) [inline paimon-python/pypaimon/read/split_read.py:170]: There is still another direct schema-manager access in DataEvolutionSplitRead._create_union_reader(): when a regular file has no write_cols, it calls self.table.schema_manager.get_schema(first_file.schema_id) to derive field ids. If first_file.schema_id is the current schema id, this bypasses _resolve_schema() and can trigger the same REST-catalog 403 on data-evolution reads. Please switch that remaining call to self._resolve_schema(first_file.schema_id) too.

**Comment by MgjLLL** [comment]
Addressed review feedback in `d43d8d99`:
1. `ManifestFileManager.read()` — added schema id short-circuit to avoid filesystem access when decoding manifest entries
2. `DataEvolutionSplitRead._create_union_reader()` — reused existing `_resolve_schema()` for the no-write_cols path

Full audit: all `schema_manager.get_schema()` calls in the read path now short-circuit when `schema_id == table_schema.id`. Remaining calls are write-path (conflict detection), branch management, and time-travel — all intentionally unchanged.

**Comment by XiaoHongbo-Hope** [comment]
+1

**Review by JingsongLi — APPROVED** [review]
+1
